allo/ir/builder.py

```diff
@@ -799,6 +799,55 @@ def build_indices(ctx, node, enable_affine=True):
 
     @staticmethod
     def build_Assign(ctx, node):
+        # Remove redundant array building
+        # pylint: disable=too-many-boolean-expressions
+        if (
```

Comment:
Add a TODO here saying that we also want to overload the normal Python operations (e.g., +/-/*) to avoid redundant tensor building